src/intrinsics.cpp

```diff
@@ -918,6 +920,180 @@ static Value *emit_intrinsic(intrinsic f, jl_value_t **args, size_t nargs,
     Value *den;
     Value *typemin;
     switch (f) {
+    HANDLE(bytevec_ref,2) {
```

Comment:
We don't want the overhead of doing a C call to pull bytes out of a buffer and we also need to be able to turn off bounds checking, which, AFAIK, there's no way to do for builtins. I'm pretty sure that Jeff wants to get rid of builtins, not add more of them. Really, the *only* reason these can't be written in pure Julia is because of the bounds checking – we can't turn it off for pure Julia code.